e3sm_diags/driver/tropical_subseasonal_driver.py

```diff
@@ -47,35 +45,28 @@ def run_diag(parameter: TropicalSubseasonalParameter) -> TropicalSubseasonalPara
     ref_data = Dataset(parameter, data_type="ref")
 
     for variable in parameter.variables:
-        test, test_start, test_end = calculate_spectrum(
-            parameter.test_data_path,
-            variable,
-            parameter.test_start_yr,
-            parameter.test_end_yr,
-        )
+        # Get test dataset
+        test_ds = test_data.get_time_series_dataset(variable, single_point=True)
```

Comment:
From what I understand, this line replaces the I/O that was being performed in `calculate_spectrum` to get the variable. This looks much cleaner now